list-agreements.lua

```diff
@@ -1,120 +1,218 @@
+    end
+    return resolution
 end
 
-function generate_output_guild(agr)
-    if is_satisfied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr).." (satisfied)")
-    elseif is_denied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr).." (denied)")
+--universal handler
+function generate_output(agr,culling,loctype)
+    local loc_name = get_location_name(agr.details[0].data.Location.tier,loctype)
+    local agr_age = get_petition_age(agr)
+    local output_str = 'Establish a '.. loc_name..' for "'..get_agr_party_name(agr)
+    
+    if loctype == df.abstract_building_type.TEMPLE then
+        output_str = output_str..'" worshiping "'..get_deity_name(agr)..',"'
+    elseif loctype == df.abstract_building_type.GUILDHALL then
+        output_str = output_str..'", a '..get_guildhall_profession(agr)..' guild,'
     else
-        print("Establish a "..get_location_name(agr).." for the "..get_guildhall_profession(agr).." guild, as agreed on "..get_petition_date(agr))
+        print("Agreement with unknown org")
+        return
+    end
+    
+    output_str = output_str..'\n\tas agreed on '..get_petition_date(agr)..'. \t'..agr_age[1]..'y, '..agr_age[2]..'m, '..agr_age[3]..'d ago'
+    if culling and (is_resolved(agr) ~= 0) then
+        return nil
+    elseif is_resolved(agr) == 1 then
+        print( output_str.." (satisfied)")
+    elseif is_resolved(agr) == 2 then
+        print(output_str.." (denied)")
+    elseif is_resolved(agr) == 3 then
+        print(output_str.." (expired)")
+    else
+        print(output_str)
     end
 end
 
-function generate_output_temple(agr)
-    if is_satisfied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for "..get_religion_name(agr)..", as agreed on "..get_petition_date(agr).." (satisfied)")
-    elseif is_denied(agr) == true then
-        print("Establish a "..get_location_name(agr).." for "..get_religion_name(agr)..", as agreed on "..get_petition_date(agr).." (denied)")
-    else
-        print("Establish a "..get_location_name(agr).." for "..get_religion_name(agr)..", as agreed on "..get_petition_date(agr))
+---------------------------------------------------------------------------
+-- Main Script operation
+---------------------------------------------------------------------------
+
+local args = {...}
+local cmd = args[1]
+local cull_resolved = true
+
+-- args handler
+if cmd then
+    if cmd == "all" then
+        cull_resolved = false
+    elseif cmd == "help" then
+        print("list-argeements: \n\t Lists outstanding Guildhall and Temple agreements in fortress mode.")
+        print("")
+        print("Supported arguments:")
+        print("\t all \n\t\t shows all agreements, past and present")
+        print("\t help \n\t\t shows this help")
```

Comment:
You can add all help text to the block at the top and then just call `print(dfhack.script_help())` here

Reply:
ah is `dfhack.script_help()` a custom function that prints comments between the `[====[`  `]====]` braces?